distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2112,6 +2144,182 @@ public void testProductDownloadWithTwoUsers() throws Exception {
                 "COMPLETE");
     }
 
+    @Test
+    public void testTwoUsersSameProductRetrySuccess() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
+        CometDClient cometDClient2 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        setupCswServerForSuccess(cometDClient1, filename);
+        setupCswServerForSuccess(cometDClient2, filename);
+
+        // Verify that we get 3 notifications: 2 retrying and 1 complete.
+        // Verify that we get 9 activity messages: started, downloading, retrying, and complete
```

Comment:
Correct me if I'm wrong, but since the csw stub server is being set up to fail twice inside your `setupCswServerForSuccess` method, the 9 probably refers to this order of activities:
`started, downloading, retrying, downloading, retrying, downloading, downloading, downloading, completed` the extra "downloading" messages are sent each time the monitor period is hit in the ResourceRetrievalMonitor and the download is still in progress.